frame/state-trie-migration/src/lib.rs

```diff
@@ -441,6 +485,10 @@ pub mod pallet {
 		/// The currency provider type.
 		type Currency: Currency<Self::AccountId>;
 
+		/// Maximal number of bytes that a key can have.
```

Comment:
Want to see a doc here about how to configure this. You can probably provide some examples of how keys are concatenations of pallet name and storage names, plus the keys of map and double map.

Reply:
Yes. I actually wanted to ask @cheme if he as an idea for a good `MaxKeyLen` value.  
I went with 128 byte, so 4 hashes maximum.  
Do we have any special cases where this would not fit or am I generally missing something?  
I will quickly check the Polkadot state for max key len as well.

Reply:
I am not a specialist of frame, but maybe there is doublekey storage or one like this, would think 3 hashes, so having 4 sounds fine.
But seeing it is only the value for the progress (one instance and one per child trie), I would certainly see no issue in taking a larger margin. But yes using a script to get it should be good (there is an rpc that count the number of V0 key value remaining, it can certainly be extended to also return the max key value).

Reply:
Turns out the maximum on Polkadot is 120 byte, quite close... no idea why.

Reply:
Added doc and bumped the default to 512.

Reply:
> Turns out the maximum on Polkadot is 120 byte, quite close... no idea why.

how do you figure this out?

Reply:
I just hacked my benchmarking since it reads the whole state. Better solution will be https://github.com/paritytech/substrate/issues/11642